src/netex_idf/offers.rs

```diff
@@ -267,13 +283,105 @@ fn enhance_with_object_code(
     enhanced_routes
 }
 
+fn arrival_departure_times(el: &Element) -> Result<(Time, Time)> {
+    fn time(el: &Element, node_name: &str) -> Result<Time> {
+        Ok(el.try_only_child(node_name)?.text().parse()?)
+    }
+    let offset: u32 = match el.try_only_child("DepartureDayOffset")?.text().parse() {
+        Ok(offset) if offset > 0 => offset,
+        Ok(offset) if offset == 0 => 0,
+        _ => 0,
+    };
+    let offset_time = Time::new(24 * offset, 0, 0);
+    let arrival_time = time(el, "ArrivalTime")?;
+    let departure_time = time(el, "DepartureTime")?;
+
+    Ok((arrival_time + offset_time, departure_time + offset_time))
+}
+
+fn boarding_type(el: &Element, node_name: &str) -> u8 {
+    el.only_child(node_name)
+        .map(|node| (node.text() == "false") as u8)
```

Comment:
A possible alternative, not sure how to it but you could use the method `.parse()` to convert from text to a `bool`.

Reply:
a bit longer but more rustic